src/datepicker/stories/DateRangePicker.stories.tsx

```diff
@@ -0,0 +1,174 @@
+import {
+  Calendar,
+  CalendarCell,
+  CalendarGrid,
+  CalendarHeader,
+  CalendarButton,
+  CalendarWeekTitle,
+  CalendarCellButton,
+  RangeCalendarStateReturn,
+} from "../../calendar";
+import {
+  useDateRangePickerState,
+  DateRangePickerInitialState,
+} from "../DateRangePickerState";
+import { stringifyDate } from "../../utils";
+
+export default {
+  title: "Component/DateRangePicker",
+} as Meta;
+
+const RangeCalendarComp: React.FC<RangeCalendarStateReturn> = state => {
+  return (
+    <Calendar
+      {...state}
+      onChange={() => console.log("change")}
+      className="calendar-range"
+    >
+      <div className="header">
+        <CalendarButton {...state} goto="previousYear" className="prev-year">
+          <DoubleChevronLeft />
+        </CalendarButton>
+        <CalendarButton {...state} goto="previousMonth" className="prev-month">
+          <ChevronLeft />
+        </CalendarButton>
+        <CalendarHeader {...state} />
+        <CalendarButton {...state} goto="nextMonth" className="next-month">
+          <ChevronRight />
+        </CalendarButton>
+        <CalendarButton {...state} goto="nextYear" className="next-year">
+          <DoubleChevronRight />
+        </CalendarButton>
+      </div>
+
+      <CalendarGrid {...state} as="table" className="dates">
+        <thead>
+          <tr>
+            {state.weekDays.map((day, dayIndex) => {
+              return (
+                <CalendarWeekTitle
+                  {...state}
+                  as="th"
+                  scope="col"
+                  key={dayIndex}
+                  dayIndex={dayIndex}
+                >
+                  <abbr title={day.title}>{day.abbr}</abbr>
+                </CalendarWeekTitle>
+              );
+            })}
+          </tr>
+        </thead>
+        <tbody>
+          {state.daysInMonth.map((week, weekIndex) => (
+            <tr key={weekIndex}>
+              {week.map((day, dayIndex) => (
+                <CalendarCell {...state} as="td" key={dayIndex} date={day}>
+                  <CalendarCellButton {...state} date={day} />
+                </CalendarCell>
+              ))}
+            </tr>
+          ))}
+        </tbody>
+      </CalendarGrid>
+    </Calendar>
+  );
+};
+
+const DateRangePickerComp: React.FC<DateRangePickerInitialState> = props => {
+  const state = useDateRangePickerState({
+    formatOptions: { month: "2-digit", day: "2-digit", year: "numeric" },
+    ...props,
+  });
+
+  return (
+    <>
+      <DatePicker className="datepicker" {...state}>
+        <div className="datepicker__header">
+          <DatePickerSegmentField
+            {...state.startSegmentState}
+            className="datepicker__field"
+          >
+            {state.startSegmentState.segments.map((segment, i) => (
+              <DatePickerSegment
+                key={i}
+                segment={segment}
+                className="datepicker__field--item"
+                {...state.startSegmentState}
```

Comment:
Since, you are using state.startSegmentState.

other state are not getting passed like isDisabled & isReadOnly etc..

Reply:
same thing for endSegmentState

Reply:
yeah right... ts is getting mad a bit tho. let's see how to fix it